flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableSource.java

```diff
@@ -126,9 +128,20 @@ public ChangelogMode getChangelogMode() {
 		return ChangelogMode.insertOnly();
 	}
 
+	@Override
+	public boolean supportsNestedProjection() {
+		// JDBC doesn't support nested projection
+		return false;
+	}
+
+	@Override
+	public void applyProjection(int[][] projectedFields) {
+		this.physicalSchema = TableSchemaUtils.projectSchema(physicalSchema, projectedFields);
```

Comment:
`selectFields` need to be updated ?

Reply:
Removed `selectFields` because we don't need it. The `physicalSchema` is the selected schema.